Sources/BranchPattern.swift

```diff
@@ -38,7 +38,7 @@ internal class BranchPattern : Pattern {
         }
     }
     
-    override func flat<T: Pattern>(_ type: T.Type) -> [T] {
```

Comment:
this unused local variable name shadows the function call later, which caused me no end of confusion